metadataproxy/roles.py

```diff
@@ -170,6 +171,29 @@ def find_container(ip):
                     log.debug(msg.format(_id, ip))
                     CONTAINER_MAPPING[ip] = _id
                     return c
+        # Try to find the container over the mesos state api and use the labels attached to it
+        # as a replacement for docker env and labels
+        if app.config['MESOS_STATE_LOOKUP']:
+            try:
+                state = json.loads(urllib2.urlopen(app.config['MESOS_STATE_URL']).read())
+                for framework in state['frameworks']:
+                    for executor in framework['executors']:
+                        for task in executor['tasks']:
+                            for status in task['statuses']:
+                                if status['state'] == 'TASK_RUNNING':
+                                    for network in status['container_status']['network_infos']:
+                                        for ip_map in network['ip_addresses']:
+                                            if ip_map['ip_address'] == ip:
+                                                if 'labels' in task:
+                                                    labels = []
+                                                    for key, val in task['labels']:
+                                                        label = '{0}={1}'.format(key, val)
+                                                        labels.append(label)
+                                                        container = {'Config': {'Env': labels, 'Labels': labels}}
+                                                        return container
+
```

Comment:
Hah, woah. This is super deeply nested. Would it be good to at least catch a KeyError at the top level here, and write out an `log.exception()`? Otherwise failures will result in 500s to the client, rather than 404.

Reply:
do you have an suggestion here how to reduce the nesting in any meaningful way ? sorry to ask but i'm not really a python guy :)

Reply:
Not really, if the data structure is this deeply nested. Mostly I was just suggesting wrapping the code in a try/except for KeyError, so that if it errors, you can log the exception, and return `None`, rather than letting an exception propagate to gunicorn, which may lose some context, and will return a 500 to the client.